core/governance/checkpoint.go

```diff
@@ -73,11 +78,14 @@ func (e *Engine) Load(ctx context.Context, data []byte) error {
 
 		switch prop.Terms.Change.GetTermType() {
 		case types.ProposalTermsTypeNewMarket:
+			et := &enactmentTime{}
 			// if the proposal is for a new market we want to restore it such that it will be in opening auction
 			if p.Terms.EnactmentTimestamp <= now.Unix() {
 				prop.Terms.EnactmentTimestamp = now.Add(duration).Unix()
+				et.lastUpdate = int64(duration)
 			}
-			toSubmit, err := e.intoToSubmit(ctx, prop)
+			et.current = prop.Terms.EnactmentTimestamp
+			toSubmit, err := e.intoToSubmit(ctx, prop, et)
```

Comment:
when we're restoring from a checkpoint, I don't think we should be validating the oracle termination timestamp vs enactment time or vs anything else. If we're restoring the market, it means it was enacted already, meaning it was valid at that time. If the oracle termination time is earlier than now or than the new enactment time (i.e. now + duration) it only means that it should fire immediately to terminate the market. 

@wwestgarth @jeremyletang @EVODelavega

Reply:
I feel like it should go straight to terminated too, but then what value is there in even restoring it.

I'm also now wondering what happens to markets that have been terminated+settle during a checkpoint reload. Because they are removed from the execution engine but not removed as enacted proposals in governance. Do those get restored and brought back to life??

Reply:
re your first point - if it's time, it will get immediately terminated so we don't need to worry or care about it. 

re your second point, you're right. The problem is that markets are not first class citizens in checkpoints, so when we save proposals in checkpoint we have no idea what the market state was at the time of the checkpoint, so I believe when its proposal is restored the market is reinstated in opening auction. So even if only terminated, or terminated and settled will reopen from scratch. 

Slightly orthogonally I'm wondering if checkpoints and LNL would live much longer (or at all) after we have the protocol upgrade?

Reply:
I thought checkpoints were the safety-net for when snapshots don't work ;)

My understanding of how the code is right now is that any restored time-triggered-oracle with a timestamp in the past will pass the validation by the timestamp-shifting, and then be ignored until the market's opening auction ends. Which maybe isn't ideal, but it doesn't sound like any of the market restoring is really working properly.

Reply:
re the future of LNL I think it's something to clarify with @barnabee and @davidsiska-vega.

re termination - my point is why do this shifting at all and do the opening auction vs terminate straight away?

Reply:
@ze97286 , @wwestgarth ,
Applied a change that will make the restored market skip the check as:
https://github.com/vegaprotocol/vega/pull/5668/commits/c418a28e496ffc53b057263de01569e7b7527bb1#diff-92b369672ec5368c6bacbb63f9c3717a133c96f06f473e577ca711abefbcbe23R85
So if we did not set the `ShouldNotVerify` value to true -> verify it